src/strategies/DSL/conditions/priceCondition.ts

```diff
@@ -1,11 +1,51 @@
-import { PrimitiveArgs } from '@brinkninja/types'
-import { PriceConditionArgs } from '@brink-sdk'
+import { PrimitiveArgs, TokenArgs } from '@brinkninja/types'
+import { PriceConditionArgs, UniV3Twap, Token } from '@brink-sdk'
 
-function priceCondition (def: PriceConditionArgs): PrimitiveArgs[] {
+const TIME_INTERVAL = BigInt(1000)
+const OPERATOR_LT = 'lt';
+const OPERATOR_GT = 'gt';
 
-  // TODO: implement
+function priceCondition ({
+  operator,
+  tokenA,
+  tokenB,
+  price
+}: PriceConditionArgs): PrimitiveArgs[] {
+  const twap = new UniV3Twap({
+    tokenA: new Token(toTokenArgs(tokenA)),
+    tokenB: new Token(toTokenArgs(tokenB)),
+    interval: TIME_INTERVAL,
+  })
 
-  return []
+  const oracle = {
+    address: twap.address,
+    params: twap.params,
+  }
+
+  switch (operator) {
+    case OPERATOR_LT:
+    return [{
+      functionName: 'requireUint256LowerBound', 
+      params: {
+        oracle: oracle,
+        lowerBound: price,
+      }
+    }]
+    case OPERATOR_GT:
+      return [{
+        functionName: 'requireUint256UpperBound',
```

Comment:
to support this, we actually need another class in `/strategies/Primitives/` called `RequireUint256UpperBound`. Maybe you could add that in this PR? basically it's the same exact thing as https://github.com/brinktrade/brink-sdk/blob/v5/src/strategies/Primitives/RequireUint256LowerBound.ts so you could just copy that. will also need to be added to `createPrimitive()` function here https://github.com/brinktrade/brink-sdk/blob/41c34a0b1bae8d5ad08314a7437de78fdda2af8b/src/internal/createPrimitive.ts

Reply:
done